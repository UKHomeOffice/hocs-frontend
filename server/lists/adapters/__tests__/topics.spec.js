const topicsAdapter = require('../topics');

const mockLogger = {
    debug: () => { },
    info: () => { },
    warn: () => { },
    error: () => { }
};

describe('Topic Adapter', () => {
    it('should transform and sort topic data', async () => {
        const mockData = {
            parentTopics: [
                {
                    options: [
                        { label: 'b', value: 2 },
                        { label: 'a', value: 1 },
                        { label: 'c', value: 3 }
                    ]
                }
            ]
        };

        const results = await topicsAdapter(mockData, { logger: mockLogger });

        expect(results).toBeDefined();
        expect(results).toMatchSnapshot();
    });
});