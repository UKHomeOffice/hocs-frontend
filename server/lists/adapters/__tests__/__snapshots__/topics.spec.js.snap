// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Adapter should transform and sort topic data 1`] = `
Array [
  Object {
    "options": Array [
      Object {
        "label": "a",
        "value": 1,
      },
      Object {
        "label": "b",
        "value": 2,
      },
      Object {
        "label": "c",
        "value": 3,
      },
    ],
  },
]
`;
